code/game/machinery/tcomms/relay.dm

```diff
@@ -0,0 +1,179 @@
+/**
+  * # Telecommunications Relay
+  *
+  * Extends the reach of telecomms to the z-level it is built on
+  *
+  * Relays themselves dont do any processing, they just tell the core that this z-level is available in the tcomms network.
+  */
+/obj/machinery/tcomms/relay
+	name = "Telecommunications Relay"
+	desc = "A large device with several radio antennas on it."
+	icon_state = "relay"
+	/// The host core for this relay
+	var/obj/machinery/tcomms/core/linked_core
+	/// ID of the hub to auto link to
+	var/autolink_id
+	/// Is this linked to anything at all
+	var/linked = FALSE
+	/// Is this link invisible on the hub?
+	var/hidden_link = FALSE
+
+/**
+  * Initializer for the relay.
+  *
+  * Calls parent to ensure its added to the GLOB of tcomms machines, before checking if there is an autolink that needs to be added.
+  */
+/obj/machinery/tcomms/relay/Initialize(mapload)
+	. = ..()
+	if(mapload && autolink_id)
+		return INITIALIZE_HINT_LATELOAD
+
+/**
+  * Descrutor for the relay.
+  *
+  * Ensures that the machine is taken out of the global list when destroyed, and also removes the link to the core.
+  */
+/obj/machinery/tcomms/relay/Destroy()
+	Reset()
```

Comment:
which is especially bad that this is part of the whole GC process.

Reply:
Fixed